src/plugins/azure/ssh.ts

```diff
@@ -45,31 +54,74 @@ export const azureSshProvider: SshProvider<
 
   propagationTimeoutMs: PROPAGATION_TIMEOUT_LIMIT_MS,
 
-  // TODO: Implement
+  // TODO: Determine if necessary
   preTestAccessPropagationArgs: () => undefined,
 
-  // TODO: Determine if necessary
+  // Azure doesn't support ProxyCommand, as nice as that would be. Yet.
   proxyCommand: () => [],
 
   // TODO: Determine if necessary
   reproCommands: () => undefined,
 
-  // TODO: Placeholder
+  setup: async (request) => {
+    // TODO(ENG-3129): Does this specifically need to be the subscription ID for the Bastion?
+    await azLogin(request.subscriptionId); // Always re-login to Azure CLI
+
+    const { path: keyPath, cleanup: sshKeyPathCleanup } =
+      await createTempDirectoryForKeys();
+    await generateSshKeyAndAzureAdCert(keyPath);
+
+    const { killTunnel, tunnelLocalPort } =
+      await trySpawnBastionTunnel(request);
+
+    const sshPrivateKeyPath = path.join(keyPath, AD_SSH_KEY_PRIVATE);
+    const sshCertificateKeyPath = path.join(keyPath, AD_CERT_FILENAME);
```

Comment:
We'd probably want to clean up the directory if the next step fails.

```suggestion
try {
    const { killTunnel, tunnelLocalPort } =
      await trySpawnBastionTunnel(request);

    const sshPrivateKeyPath = path.join(keyPath, AD_SSH_KEY_PRIVATE);
    const sshCertificateKeyPath = path.join(keyPath, AD_CERT_FILENAME);
 } finally {
   await sshKeyPathCleanup()
 } 
```

Reply:
Done, but implemented as a `try`/`catch` instead of `try`/`finally`, since we don't want to trigger the key path cleanup here if we're able to spawn the tunnel successfully.

Reply:
Oh right, finally is wrong here!